pkg/convert/tf_scopes.go

```diff
@@ -225,6 +225,8 @@ func (s *scopes) getOrAddPulumiName(path, prefix, suffix string) string {
 
 // Given a fully typed path (e.g. data.simple_data_source.a_field) returns the final part of that path
 // (a_field) and the either the Resource or Schema, and SchemaInfo for that path (if any).
+//
+// Can return PathInfo{} if the final part cannot be resolved. The caller is then responsible for handling that case.
```

Comment:
I know it's kinda Go-ish to just throw zero values around like this but I feel like this ought to be more explicit. Very easy for a callsite to look at this and think it's always going to get a sensible result back. Maybe `*PathInfo` or `(PathInfo, bool)`?

Reply:
Updated to `(PathInfo, bool)`